polyfactory/factories/base.py

```diff
@@ -661,6 +664,34 @@ def get_model_fields(cls) -> list[FieldMeta]:  # pragma: no cover
         """
         raise NotImplementedError
 
+    @classmethod
+    def get_factory_fields(cls) -> list[tuple[str, Any]]:
+        factory_fields = cls.__dict__.items()
+        return [
+            (field_name, field_value) for field_name, field_value in factory_fields
+            if not (field_name.startswith("__") or field_name == "_abc_impl")
+        ]
+
+
+    @classmethod
+    def _check_declared_fields_exist_in_model(cls):
+        model_fields_names = [field_meta.name for field_meta in cls.get_model_fields()]
+        factory_fields = cls.get_factory_fields()
+
+        for field_name, field_value in factory_fields:
+            if field_name in model_fields_names or inspect.isfunction(field_value):
+                continue
+            else:
+                error_message = (
+                    f"{field_name} is declared on the factory {cls.__name__}"
+                    f" but it is not part of the model {cls.__model__.__name__}"
+                )
+                if isinstance(field_value, (Use, PostGenerated, Ignore, Require)):
+                    raise ConfigurationException(error_message)
+                else:
+                    warnings.warn(ConfigurationWarning(error_message))
```

Comment:
I don't think that this warning should be there. If I'm understanding it correctly, this prevents users from creating any class attributes they might want for some reason without facing a warning. The following is a bit of a contrived example, but it would create a warning when it really should not.

```python
@dataclass
class Foo:
    foo: int

class FooFactory(DataclassFactory[Foo]):

    some_attribute: int
    
    @classmethod
    def foo(cls) -> int:
        return cls.some_attribute * 5
```

Reply:
This is why I completely ignored classmethod for now. It will only « clash » with regular attributes which IMO is acceptable as it’s only a warning. It can also be disabled if someone doesn’t want those to be triggered.

Reply:
Hm..I have to disagree. I think that users should be able to define any attribute they want without it triggering a warning. I'm not sure I'm a fan of limiting what the user can and cannot define.

That being said, in the next major version I think we should go with your idea of not allowing any random classmethod to be used to get the value for a field. Instead, users would have to use the `Use` construct and we could create a `@use` decorator like the `@post_generated` decorator that exists currently.

Reply:
Fair enough 👌🏻 so to summarize we should only check for known construct? and do nothing for any other case like regular attribute and classmethod, correct?

Reply:
Yupp.

Reply:
another idea I had btw is to define a decorator to silence the warning, something like ˋnon_model_attribute`.

I like your proposal better but this one would avoid any breaking change really as I don’t consider the warning as breaking per se.